lib/livestatus/livestatuslogutility.cpp

```diff
@@ -80,11 +80,24 @@ void LivestatusLogUtility::CreateLogCache(std::map<time_t, String> index, Histor
 
 	/* m_LogFileIndex map tells which log files are involved ordered by their start timestamp */
 	unsigned long line_count = 0;
-	for (const auto& kv : index) {
+	std::map<time_t, String>::const_iterator last = index.end();
+	--last;
```

Comment:
What if index.end() is pointing to index.begin()?

Reply:
Thank you i did not think about, it would also imply thath compat log is disabled or at least current log will be present i will placea return on index.empty at the very begin of sub

Reply:
I tested the case of no compat log against 3482dfa without hitting problem do you see any other way to have index.end() pointing to index.begin() ?